tests/test_diffusers.py

```diff
@@ -730,3 +733,344 @@ def test_no_generation_regression_upscale(self):
+
+        # Test num_images_per_prompt=1 (default) for several prompts
+        num_prompts = 3
+        images = sd_pipe([prompt] * num_prompts, num_inference_steps=2, output_type="np").images
+
+        self.assertEqual(len(images), num_prompts)
+        self.assertEqual(images[-1].shape, (64, 64, 3))
+
+        # Test num_images_per_prompt for single prompt
+        num_images_per_prompt = 2
+        images = sd_pipe(
+            prompt, num_inference_steps=2, output_type="np", num_images_per_prompt=num_images_per_prompt
+        ).images
+
+        self.assertEqual(len(images), num_images_per_prompt)
+        self.assertEqual(images[-1].shape, (64, 64, 3))
+
+        # Test num_images_per_prompt for several prompts
+        num_prompts = 2
+        images = sd_pipe(
+            [prompt] * num_prompts,
+            num_inference_steps=2,
+            output_type="np",
+            num_images_per_prompt=num_images_per_prompt,
+            ).images
+
+        self.assertEqual(len(images), num_prompts * num_images_per_prompt)
+        self.assertEqual(images[-1].shape, (64, 64, 3))
+
+    def test_stable_diffusion_xl_batch_sizes(self):
+        components = self.get_dummy_components()
+        gaudi_config = GaudiConfig()
+
+        sd_pipe = GaudiStableDiffusionXLPipeline(
+            use_habana=True,
+            gaudi_config=gaudi_config,
+            **components,
+        )
+        sd_pipe.set_progress_bar_config(disable=None)
+
+        prompt = "A painting of a squirrel eating a burger"
+
+        # Test batch_size > 1 where batch_size is a divider of the total number of generated images
+        batch_size = 3
+        num_images_per_prompt = batch_size**2
+        images = sd_pipe(
+            prompt,
+            num_inference_steps=2,
+            output_type="np",
+            batch_size=batch_size,
+            num_images_per_prompt=num_images_per_prompt,
+        ).images
+        self.assertEqual(len(images), num_images_per_prompt)
+        self.assertEqual(images[-1].shape, (64, 64, 3))
+
+        # Same test for several prompts
+        num_prompts = 3
+        images = sd_pipe(
+            [prompt] * num_prompts,
+            num_inference_steps=2,
+            output_type="np",
+            batch_size=batch_size,
+            num_images_per_prompt=num_images_per_prompt,
+            ).images
+
+        self.assertEqual(len(images), num_prompts * num_images_per_prompt)
+        self.assertEqual(images[-1].shape, (64, 64, 3))
+
+        # Test batch_size when it is not a divider of the total number of generated images for a single prompt
+        num_images_per_prompt = 7
+        images = sd_pipe(
+            prompt,
+            num_inference_steps=2,
+            output_type="np",
+            batch_size=batch_size,
+            num_images_per_prompt=num_images_per_prompt,
+        ).images
+
+        self.assertEqual(len(images), num_images_per_prompt)
+        self.assertEqual(images[-1].shape, (64, 64, 3))
```

Comment:
Seems that default images are of size 1024x1024 -->  128x128 latents. (see https://github.com/huggingface/diffusers/blob/main/docs/source/en/using-diffusers/sdxl.md#text-to-image).  Not sure how here we are working with 64x64 here unless we have a bug and SDXL pipe default is 512x512?  Let's check.

I think the recommended square size for XL is 1024x1024 (see https://stability.ai/sdxl-aws-documentation).  512x512 likely to functionally work but could yield bad looking image.   Maybe OK for CI tests.

Reply:
Good point. I copied the test model from diffusers which is using a smaller unet. I think they are using the smaller size for faster tests. https://github.com/huggingface/diffusers/blob/main/tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py#L164

Reply:
Ok so diffusers is also using wrong unet :)  In that case lets use the same, although not correct.   We could try to submit PR for fix to diffusers (or open a new issue), and if they fix, we can too.  But lets put that on back burner for now.